File: lib/java/openlinktoken-core-ai/src/main/java/org/openlinktoken/core/ai/tokens/definitions/ML1Token.java
/* SPDX-License-Identifier: MIT */
package org.openlinktoken.core.ai.tokens.definitions;

import java.util.ArrayList;

import org.openlinktoken.attributes.AttributeExpression;
import org.openlinktoken.tokens.Token;

/**
 * Represents optional ML1 token definition.
 */
public class ML1Token implements Token {
    private static final long serialVersionUID = 1L;
    public static final String TOKEN_ID = "ML1";

    private final ArrayList<AttributeExpression> definition = new ArrayList<>();

    /**
     * Return the stable registry identifier for this token definition.
     *
     * @return ML1 token identifier
     */
    @Override
    public String getIdentifier() {
        return TOKEN_ID;
    }

    /**
     * Return the attribute expressions that make up the ML1 definition.
     *
     * @return ordered ML1 attribute expressions
     */
    @Override
    public ArrayList<AttributeExpression> getDefinition() {
        return definition;
    }
}

File: lib/python/openlinktoken-core-ai/src/main/openlinktoken/core/ai/tokens/ml1_token.py
"""Optional ML1 token definition."""

from typing import List

from openlinktoken.attributes.attribute_expression import AttributeExpression
from openlinktoken.tokens.token import Token


class ML1Token(Token):
    """Represents optional ML1 token definition."""

    TOKEN_ID = "ML1"

    def __init__(self):
        """Initialize an ML1 definition with no attribute expressions."""
        self._definition: List[AttributeExpression] = []

    def get_identifier(self) -> str:
        """Return the stable registry identifier for this token definition."""
        return self.TOKEN_ID

    def get_definition(self) -> List[AttributeExpression]:
        """Return the attribute expressions that make up the ML1 definition."""
        return self._definition
